src/test/java/com/gluonhq/richtext/model/PieceTableTests.java

```diff
@@ -209,6 +210,86 @@ public void sameBlockDecorate() {
         );
     }
 
+    @Test
+    @DisplayName("Same block decorate posture")
+    public void sameBlockDecoratePosture() {
+        String text = "Original Bigger Text";
+        PieceTable pt = new PieceTable(text);
+        pt.decorate(1,2, TextDecoration.builder().fontPosture(FontPosture.ITALIC).build());
+        Assertions.assertEquals(
+                text,
+                pt.getText()
+        );
+        Assertions.assertTrue(pt.pieces.stream()
+                .filter(piece -> piece.getText().equals("r"))
+                .anyMatch(piece -> piece.getDecoration().getFontPosture() == FontPosture.ITALIC)
+        );
+    }
+
+    @Test
+    @DisplayName("Same block decorate foreground color")
+    public void sameBlockDecorateForegroundColor() {
+        String text = "Original Bigger Text";
+        PieceTable pt = new PieceTable(text);
+        pt.decorate(1,2, TextDecoration.builder().foreground(Color.AQUA).build());
+        Assertions.assertEquals(
```

Comment:
Can you keep this is one line? (same below)

Reply:
Like this:

```
Assertions.assertEquals(text, pt.getText());
```

Or do you want me to somehow merge both assertions?

Reply:
Yes, like that